schemas/Bills/tradeBill.yaml

```diff
@@ -97,92 +184,108 @@ components:
           nullable: true
         externalSystemBillId:
           type: string
-        reversalBillId:
+          nullable: true
+        externalSystemReversalBillId:
           type: string
           nullable: true
         postedDate:
           type: string
           format: date
           description: "ISO 8601 date format (YYYY-MM-DD)"
-
+          nullable: true
+          
     Project:
       type: object
       required:
-        - projectId
+        - id
         - projectName
         - lotType
-        - projectStatus
+        - status
       properties:
-        projectId:
-          type: string
-        projectName:
+        id:
+          $ref: '#/components/schemas/ID'
+        name:
           type: string
         lotType:
-          type: string
-        projectStatus:
-          type: string
+          $ref: '#/components/schemas/LotType'
+        status:
+          $ref: '#/components/schemas/ProjectStatus'
         warrantyStartDate:
           type: string
           format: date
           description: "ISO 8601 date format (YYYY-MM-DD)"
           nullable: true
-        verticalCompletionDate:
+        verticalCompleteDate:
           type: string
           format: date
           description: "ISO 8601 date format (YYYY-MM-DD)"
           nullable: true
 
-    LineItem:
+    BillLineItem:
       type: object
       required:
-        - lineId
+        - id
         - amountInCents
-        - costCodeId
-        - costCodeNumber
-        - costClassification
+        - costCode
       properties:
-        lineId:
-          type: string
+        id:
+          $ref: '#/components/schemas/ID'
         amountInCents:
           type: integer
-        paidInCents:
+        paidInCents: # i'm not sure this should be included, it is a prorated field based off the total in the bill when the payment info should flow from fdi to blueprint
           type: integer
-        costCodeId:
-          type: string
-        costCodeNumber:
-          type: string
-        costClassification:
-          type: string
-        costDivision:
-          type: string
-          nullable: true
-        costCodeVersion:
-          type: integer
-        sourceId:
-          type: string
-        description:
-          type: string
+        costCode:
+          $ref: '#/components/schemas/CostCode'
 
     Approval:
       type: object
       required:
         - approvalStatus
       properties:
-        approvalStatus:
-          type: string
-        approvalDate:
+        id:
+          $ref: '#/components/schemas/ID'
+        status:
+          $ref: '#/components/schemas/ApprovalStatus'
+        statusUpdatedAt:
```

Comment:
this is technically a property of each underlying approver, but i think it would be safe to calculate as `approvers.max((a) => a.statusUpdatedAt)`.  We don't store it at the approval level, and also don't have separate fields for approved and rejected.

also we reserve the suffix `...At` for datetimes and `...Date` or `...On`  for plain dates